opm/parser/eclipse/EclipseState/Tables/JFuncTable.hpp

```diff
@@ -0,0 +1,115 @@
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ OPM is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef OPM_PARSER_ECLIPSE_ECLIPSESTATE_TABLES_JFUNCTABLE_HPP_
+#define OPM_PARSER_ECLIPSE_ECLIPSESTATE_TABLES_JFUNCTABLE_HPP_
+
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+
+#include <boost/multi_array.hpp>
+
+#include <iostream>
+
+namespace Opm {
+
+class DeckKeyword;
+
+class JFuncTable
+{
+public:
+
+    enum Flag {
+        BOTH, WATER, GAS
+    };
+    enum Direction {
+        XY, X, Y, Z
+    };
+
+    inline void init(const DeckKeyword& kw) {
+        const auto& rec = kw.getRecord(0);
+        if (rec.hasItem("FLAG")) {
+            const auto& kw_flag = rec.getItem("FLAG").get<std::string>(0);
+            if (kw_flag == "WATER")
+                flag = WATER;
+            else if (kw_flag == "GAS")
+                flag = GAS;
+        }
+
+        if (flag != WATER)
+            goSurfaceTension = rec.getItem("GO_SURFACE_TENSION").get<double>(0);
+
+        if (flag != GAS)
+            owSurfaceTension = rec.getItem("OW_SURFACE_TENSION").get<double>(0);
+
+        if (rec.hasItem("ALPHA_FACTOR"))
+            alphaFactor = rec.getItem("ALPHA_FACTOR").get<double>(0);
+        if (rec.hasItem("BETA_FACTOR"))
+            betaFactor  = rec.getItem( "BETA_FACTOR").get<double>(0);
+
+        if (rec.hasItem("DIRECTION")) {
+            const auto kw_dir = rec.getItem("DIRECTION").get<std::string>(0);
+            if (kw_dir == "X")
+                direction = X;
+            else if (kw_dir == "Y")
+                direction = Y;
+            else if (kw_dir == "Z")
+                direction = Z;
+        }
+
+    }
+
+    inline double getAlphaFactor() const {
+        return alphaFactor;
+    }
+
+    inline double getBetaFactor() const {
+            return betaFactor;
+    }
+
+    inline double getgoSurfaceTension() const {
+        if (flag == WATER)
+            throw std::invalid_argument("Cannot get gas-oil with WATER JFUNC");
+        return goSurfaceTension;
+    }
+
+    inline double getowSurfaceTension() const {
+        if (flag == GAS)
+            throw std::invalid_argument("Cannot get oil-water with GAS JFUNC");
+        return owSurfaceTension;
+    }
+
+    inline const Flag& getJFuncFlag() const {
+        return flag;
+    }
+
+    inline const Direction& getDirection() const {
+        return direction;
+    }
+
+private:
+    Flag      flag = BOTH;             // JFUNC flag: WATER, GAS, or BOTH.  Default BOTH
```

Comment:
I would prefer if these default values were removed - they are already specified in the Json configuration, and if the values are requested unconditionally from the deck we will correctly get either the default or the value entered explicitly.